### PR TITLE
Bumped minor version from 1.4.0 to 1.5.0 on master, updated License file

### PR DESCRIPTION
**Pull request by srochel**
## Description ##
Bumped minor version from 1.4.0 to 1.5.0 on master

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [X ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [X ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by srochel** [comment]
@nswamy @gigasquid @sergeykolychev @szha @lebeg - please review and approve.

File: contrib/clojure-package/README.md [hunk 1]
```diff
@@ -147,7 +147,7 @@ The first step is to recursively clone the MXNet repository and checkout the des
   ```bash
   git clone --recursive https://github.com/apache/incubator-mxnet.git ~/mxnet
   cd ~/mxnet
-  git checkout tags/1.4.0 -b my_mxnet  # this is optional
+  git checkout tags/1.5.0 -b my_mxnet  # this is optional
```
Comment (gigasquid) [inline contrib/clojure-package/README.md]: this needs to be 1.4.0

File: contrib/clojure-package/README.md [hunk 2]
```diff
@@ -176,7 +176,7 @@ The outcome of this step will be a shared library `lib/libmxnet.so` that is used
 
 #### Building the Clojure jar
  
-- Enter the `contrib/clojure` directory and edit the `project.clj` file. Add the Scala jar that was just created and installed, e.g., `[org.apache.mxnet/mxnet-full_2.11-osx-x86_64-cpu "1.4.0-SNAPSHOT"]`, to the `:dependencies`.
```
Comment (gigasquid) [inline contrib/clojure-package/README.md]: This one is ok as we are building it

File: contrib/clojure-package/README.md [hunk 3]
```diff
@@ -105,9 +105,9 @@ brew install opencv
 - Create a new project with `lein new my-mxnet`
 - Edit your `project.clj` and add one of the following entries to `:dependencies`, based on your system and the compute device you want to use:
 
-  - `[org.apache.mxnet.contrib.clojure/clojure-mxnet-linux-cpu "1.4.0"]`
-  - `[org.apache.mxnet.contrib.clojure/clojure-mxnet-linux-gpu "1.4.0"]`
-  - `[org.apache.mxnet.contrib.clojure/clojure-mxnet-osx-cpu "1.4.0"]`
+  - `[org.apache.mxnet.contrib.clojure/clojure-mxnet-linux-cpu "1.5.0"]`
```
[the remark this answers is not in the document]
Comment (srochel) [reply contrib/clojure-package/README.md]: only the CPU version or cpu, gpu and macos version need to be 1.4.0?

File: contrib/clojure-package/README.md [hunk 4]
```diff
@@ -116,20 +116,20 @@ After making this change and running `lein deps`, you should be able to run exam
...
 - Install additional dependencies as described in [the corresponding section for Option 1](#installing-additional-dependencies),
-- Recursively clone the MXNet repository and checkout the desired revision. Here we assume the `1.4.0` tag and a clone into the `~/mxnet` directory:
+- Recursively clone the MXNet repository and checkout the desired revision. Here we assume the `1.5.0` tag and a clone into the `~/mxnet` directory:
 
   ```bash
   git clone --recursive https://github.com/apache/incubator-mxnet.git ~/mxnet
   cd ~/mxnet
   git tag --list  # Find the tag that matches the Scala package version
-  git checkout tags/1.4.0 -b my_mxnet
+  git checkout tags/1.5.0 -b my_mxnet
```
[the remark this answers is not in the document]
Comment (srochel) [reply contrib/clojure-package/README.md]: Neither 1.4.0 nor 1.5.0 tag exists. Why would you want to check out a different version?

File: contrib/clojure-package/README.md [hunk 5]
```diff
@@ -116,20 +116,20 @@ After making this change and running `lein deps`, you should be able to run exam
...
   git submodule update --init --recursive
   cd contrib/clojure
   ```
 
 - Edit `project.clj` to include the desired Scala jar from Maven:
 
-      [org.apache.mxnet/mxnet-full_2.11-linux-x86_64-cpu "1.4.0”]
+      [org.apache.mxnet/mxnet-full_2.11-linux-x86_64-cpu "1.5.0”]
```
[the remark this answers is not in the document]
Comment (srochel) [reply contrib/clojure-package/README.md]: should this point to the latest released version, i.e. 1.3.0 today and 1.3.1 after the patch is released? Or should it point to nightly build?

File: contrib/clojure-package/README.md (hunk 1 above)
Comment (srochel) [reply contrib/clojure-package/README.md]: Neither 1.4.0 nor 1.5.0 tag exists. 1.4.0 tag will point to a release in v1.4.x branch. Which should be the correct tag? Should it point to a release tag?

Comment (gigasquid) [reply contrib/clojure-package/README.md]: It should point to the tag that corresponds to the jar on maven and the jars on maven are only for the last release. I'm assuming that it will be the 1.4.0 tag when this branch is merged (at least that's what I've been doing).  For reference when I do `git tag --list` I see:

```
1.3.1
1.3.1.rc0
```
I'm not exactly sure of the timing of when this PR is merged though .. is that a correct assumption?

File: contrib/clojure-package/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (gigasquid) [reply contrib/clojure-package/README.md]: It should point to the latest release on maven for the clojure jars. It would be great for the users if we can reliable tell that at the time of the merge, but if this process gets to be too confusing we can always reword it to stay explicitly that it should be the latest release https://search.maven.org/search?q=clojure%20mxnet - Let me know your thoughts.

File: contrib/clojure-package/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (gigasquid) [reply contrib/clojure-package/README.md]: see answer below..

File: contrib/clojure-package/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (gigasquid) [reply contrib/clojure-package/README.md]: all the versions osx cpu, linux cpu, and linux gpu need to be the latest version out on maven https://search.maven.org/search?q=clojure%20mxnet at the time this PR gets merged. That timing is the thing I don't well understand. I am making an assumption at the time of the merge there will be a 1.4.0 release done and the jars will be out there.

[the remark this answers is not in the document]
Comment (srochel) [reply contrib/clojure-package/README.md]: This PR is increasing the version on master to differentiate between v1.4.x branch. I would like to merge the PR asap, so we don't have same version on master and v1.4.x branch. v1.4.0 will become the first release on v1.4.x branch. If everything goes to plan, the release should be done before Xmas. The latest tag is now 1.3.1. Should all references point to this tag?

**Comment by srochel** [comment]
@gigasquid - I see you opened PR 13507 to update the README.md for clojure package. Is it ok to merge this PR and you will follow with update on contrib/clojure-package/README.md? The v1.4.x branch will require an update as well. Please advice.

**Comment by gigasquid** [comment]
@srochel - Sure go ahead and merge this PR first - I will either close mine or update it based on these changes 😸

**Comment by TaoLv** [comment]
@srochel @sergeykolychev Seems this PR was messed up by code rebasing and many non-related code was merged into master. Please take a look at the changed files. Thanks.

**Comment by srochel** [comment]
@TaoL - sorry about this. Looks like I made a mistake with the PR and see that mklddn related changes got included. I need to check my setup. 
What do you suggest is the best way to correct? Should the PR be reverted or is there a better way to exclude the mkldnn related changes? Feel free to revert the PR or advice on best solution. I will check in the morning PST.

**Comment by TaoLv** [comment]
@srochel I will revert this PR on the master branch to mitigate issues. Feel free to submit new PR for your changes. Sorry for any inconvenience.
